xl-objects.go

```diff
@@ -249,21 +250,47 @@ func (xl xlObjects) getObjectInfo(bucket, object string) (ObjectInfo, error) {
 		fi.ModTime = info.ModTime
 		fi.MD5Sum = info.MD5Sum
 	}
-	contentType := "application/octet-stream"
-	if objectExt := filepath.Ext(object); objectExt != "" {
-		content, ok := mimedb.DB[strings.ToLower(strings.TrimPrefix(objectExt, "."))]
-		if ok {
-			contentType = content.ContentType
+
+	metadata := make(map[string]string)
+	offset := int64(0) // To read entire content
+	r, err := xl.storage.ReadFile(bucket, pathJoin(object, "meta.json"), offset)
+	if err == nil {
+		// FIXME: Ignore err as multipart does not have meta data now.
```

Comment:
Move this code out, multipart has metadata now - rebase the changes. 

Also easier is to use json.NewDecoder(r) like this.